### DISCUSSION TITLE
Work for AVANTIS

### DISCUSSION THREAD

**Discussion by Pocky0013** [togrupe/dlive-midi-tools#20] [Show and tell]
Thank you very much for this great tool. 
I just did several tests with my Avantis and overall everything works. 

The program momentarily crashes when it tries to send phantom and pad information to DX3 because it is not present on the AVANTIS. But that does not preclude using the software. 

And the Reaper template is just magic! Thank you so much

**Comment by togrupe** [comment]
Hey @Pocky0013

thanks for the nice feedback. 

Till last Friday I didn´t even know that it works with Avantis ;) 

I will try the support Avantis "officially" probably with a dropdown (dlive | avantis) because obviously its almost the same interface. 

Hope that from the midi interface perspective its also possible to configure the Slink port as well. Let me check. 

If you have any ideas to improve the workflow, please let me know. 

Tobi

PS: How did you get informed about this tool?

**Reply by Pocky0013** [reply]
from what I have been able to test. The Slink control is already working. With the DX1 column of the excel file, I have good control of the pad and the 48V on my 32 channels. I have an AR2412 and an AR0804 linked.
On the other hand, the software displays for about 10 seconds: "Does not respond" because I think that since there is no DX3, the noon command does not pass. But the process ends anyway.

As you say, I think a simple "DLive or Avantis" menu that allows you to deactivate the DX3 sync would be perfect

I see 2 points that improved the process.
1.  It is that the reaper file is saved in the same place as the excel file and not in the software folder. This allows you to have configuration sheets in folders for each client.
2. That the repaer file partly takes over the name of the execl file.

Example: PatchClient1.xls = PatchClient1_Recording-template.rpp

this will allow you to directly open several reaper configs without having to rename them each time.

Another interesting point would be having to influence the routing directly. Often I have 1 or 2 tracks using the same preamp.
In my patch sheets, I have a track column and a preamp column. Normally track 1 is always on pream 1, and so on.
Without going into a complex routing, if we could simply choose the preamp number for each track, that would be great. But it's really something not essential


I think I must have seen a Facebook post on Friday asking a question about the tools and I thought it would be really cool. I always do patch sheet on excel upstream and always I had to manually put the information in the console.

I give you a link on my excel sheet that I complicated with yours by adding formulas. There are still small things to do, but it gives an idea of ​​the possibilities (https://www.swisstransfer.com/d/f7ace871-511e-496f-9c75-6fc600112948)

**Reply by togrupe** [reply]
Hi @Pocky0013 

Thanks for your input! 

1.  Yes I think, "ignore" the DX3 in case of Avantis should be simple. We only should think about a proper column name because its now basically DX1 and SLink (in case of Avantis)

2. Regarding Reaper Templates:

Did you use the "binary (download) version" or did you run it manually from the python console? macOS or Windows? 

I like the idea very much, you proposed. Makes totally sense.

> Example: PatchClient1.xls = PatchClient1_Recording-template.rpp 

the next version will have that feature. You want the Reaperfile be generated right next to the input file correct?

3. What do you mean with Routing? The assignment to busses or the IO Routing (Socket to DSP-Channel), or do you mean the Reaper Routing within the Reaper template?  

Can you explain this a bit more detailed? 

The problem with the "Socket to DSP Channel" Routing is, that this is currently (to my knowledge) not supported from the midi/tcp interface. I would love to have this as well. Assignment to busses, can be done. 

Tobi

**Reply by Pocky0013** [reply]
I used the binary version for windows. I'm not familiar with the console so the binary file helped me a lot.
As we must select the location of the XLS input file, it would actually be good for the repaer file to be saved in the same places. This way I can have a single software folder and manage my patch sheets in folders according to my convenience.

Regarding the routing I was actually thinking of the internal routing of the console (socket to DSP). But indeed if this is not "yet" possible in midi/tcp, this will pose a problem. Maybe in the new A&H firmware version. 
The reaper routing is perfect for me as well. Using DanteVirtual sound Card, the inputs and outputs in reaper have been directly configured correctly.

**Reply by togrupe** [reply]
@Pocky0013 
Fully agree with the file location for reaper file. 

Where do you come from?

**Reply by Pocky0013** [reply]
I live in Switzerland on the French-speaking side near Montreux.
I can't help you with coding, but I do IT maintenance and I'm used to testing systems. Do not hesitate to ask me you need me to test things in beta on the Avantis

**Comment by togrupe** [comment]
@Pocky0013 

Do you know how many input sockets can be available on the Slink Port? 128 including the DX1/2
And how does the patching work if you have a GX48/16 and DX1/2 available?

1-48 (GX), 65-96 (DX1) ,97-128 (DX2)?

And how did you make it work in your Stagebox combination

**Reply by Pocky0013** [reply]
from what I have seen and understood. On the Slink you can only put 48/16.
Like the basic avantis only 1 Slink port and no GX. If a second line is possible, a GX expansion card must be installed. In this case, I don't know if the routing will be like with a standard expansion card (for example my dante card) via the routing of Port I/O 1 or 2, or if it will be smarter than that and will recognize the hardware through the expansion card.

in my configuration, I leave the console via the Slink to go to an AR24/12 and from the AR24/12, I use the "extension" port of it to go to an AR08/04. But in this configuration, I am at the max of capacity. It is not possible to cascade two AR24/12
So in the excel file, I stopped at the "socket 32" level and I have no way to test more extenders

**Comment by togrupe** [comment]
@Pocky0013 

For the 1. feature: "Generate Reaper file right next to input file with prefix", I have already a version. Would you like to test it?

Regarding the 2. feature: "Slink Pad / 48V support on Avantis" . What do you think about the following. 

<img width="585" alt="Bildschirmfoto 2023-04-25 um 09 37 49" src="https://user-images.githubusercontent.com/89751618/234207470-4b698abc-aab7-4980-8b6a-4a4945a0d9d6.png">


Unfortunately, Officially due to midi/tcp document on Avantis the "48V and pad" feature is not supported but obviously it seem to work. So we should be careful.

**Reply by togrupe** [reply]
I found the following, regarding GX4816 and Slink patching

<img width="1124" alt="Bildschirmfoto 2023-04-25 um 09 46 02" src="https://user-images.githubusercontent.com/89751618/234209717-5d45c706-fd8d-4d59-98fe-ffa55ed7333e.png">

**Reply by togrupe** [reply]
the dlive/Avantis Supported UI could look like that...what do you think?

<img width="701" alt="Bildschirmfoto 2023-04-25 um 09 57 31" src="https://user-images.githubusercontent.com/89751618/234212456-c99f33b4-629f-4644-9281-38e882f24e92.png">

**Reply by Pocky0013** [reply]
for the excel file I will find it more logical to either sort by type of port or by function. Personally I find it clearer when it's sorting by type of port.
![Capture](https://user-images.githubusercontent.com/111347509/234214222-931e9705-39a8-4543-a444-ee07778451c1.JPG)


the interface you propose with the drop-down list is great

Excellent for the diagrams you found, I went through the documentation this morning and I was looking for this ! 
I'm not in the office in front of the console today, but I can test the new version tomorrow with no problem. Can you make me a binary compilation for windows?

**Reply by togrupe** [reply]
@Pocky0013 

Interesting idea.

The order of the columns doesn't matter, in theory everyone can sort it, as he likes it. The only issue could be the first line. I am not sure if the spreadsheet loader library can load it, because the lib searches for the headline. e.g "Local Phantom".

**Reply by Pocky0013** [reply]
@togrupe  The first line with the titles is not necessary if you can sort the columns as you want and it has no impact on the script. the colors already gives something clear access

**Comment by togrupe** [comment]
@Pocky0013 

This is the feature 1: "Generate Reaper file right next to input file with prefix" version.

http://wp1054826.server-he.de/downloads/dlive-midi-tools/v2_2_0-alpha-SNAPSHOT/windows/dmt-v2_2_0-alpha-SNAPSHOT-windows.zip

**Reply by Pocky0013** [reply]
NICE !!!! 
I just did the reaper part because I don't have the console with me now. But the file is well named and correctly saved in the folder entered in the utility. nice job dude

**Comment by togrupe** [comment]
Just did a small test...reorganizing the columns works

<img width="655" alt="Bildschirmfoto 2023-04-25 um 15 06 07" src="https://user-images.githubusercontent.com/89751618/234285763-483308ec-713a-4683-8595-d0d1e06809d6.png">

**Reply by Pocky0013** [reply]
I also tested and it works

**Comment by togrupe** [comment]
Hey @Pocky0013 

Feature 2: Slink Pad / 48V support on Avantis - has now also a release candidate. 

http://wp1054826.server-he.de/downloads/dlive-midi-tools/v2_2_0-alpha2-SNAPSHOT/dmt-v2_2_0-alpha2-SNAPSHOT-windows.zip

It has one technical limitation, due to the used midi library. We can only confgure the slink ports 1-64. This means when using a GX4816 only local ports (1-48) and DX1 (64-95) are writable.

i also reorganized the excel sheet tab "48V & Pad" by ports.

I am curiuous about your feedback.

**Reply by togrupe** [reply]
An additional hint...  Keep in mind to use the latest spreadsheet. Version 3 (in Tab "Misc"), Its part of the zip file.

Otherwise you should see a Warning.

**Reply by Pocky0013** [reply]
Testing one by one FOR AVANTIS
- [x] Name
- [x] Color
- [x] Mute
- [x] 48V Phantom local Avantis 1 to 12
- [x] PAD local Avantis 1 to 12
- [x] 48V Phantom Slink (AR2416 + AR0804) (socket number 1 to 32)
- [x] PAD Slink (AR2416 + AR0804) (socket number 1 to 32)
Testing everything together

- [x] ALL function (Write to console) takes 58 seconds when there is need to change all information

**Generate Repaer recoding session**

- [ ] Generate Reaper Recoding session

I am encountering a problem. The process gets stuck at 30% and does not complete. However it looking like the reaper file is well generated with the right tracks, color and routing...

**Improvement proposal**

- as there are 128 socket in the excel file, that's a lot of empty track in my repaer file. It would be nice to be able to choose how many tracks you want in the template. We could caller on the possibilities of the dantes cards namely: 32 - 48 - 64 or 128 tracks.
- This will add a small drop-down menu to make this choice. What do you think about it ?
- would it be possible to "save" the choices of the IP address and the console for the next opening of the utility?

- the button of the drop-down list looks strange to me
![Capture](https://user-images.githubusercontent.com/111347509/234791902-2df47b31-acf8-4d39-83ca-35b114e23b40.JPG)
but that work !


**_### it is truly a fantastic job you have done. thank you so much_**

**Comment by togrupe** [comment]
Wow... fantastic feedback. Thank you so much.

1. Regarding the range writing.. 

this is implicitly already possible...
you can delete the rows in the spreadsheet.
for example have spread sheet like this..
1-16
19-20
25-32
41-48

the only limitation is, that there should be no „empty lines“ between the "blocks"

I do this very often to speed up the writing process. The reason why it takes so long it that the dlive system take a few ms to process the data, so we have some „sleeps" in the code. 

Your idea of defining a start and stop channel in a dropdown is cool. In this case all channels can be still available within the spreadsheet and you just filter on the fly.


3. Regarding the optical style of the dropdown buttons 

this is the default for windows. I also don’t like it ;) I use mainly my Mac do drive the development, and there it looks much more "nicer" :)
but nevertheless I will try to search for an alternativ GUI style for windows.  


4. IP Topic: 

When you run the tool based on python you can change the "ip" in a configuration file.

Saving current ip-address from GUI Button is a great idea.. could also be part of the next release. I also thought about a „Director“ Button, that override the ip with 127.0.0.1 and if you unclick it, it switches back the the previous set ip.

5. Partial freeze of the GUI 
I only saw this once in the windows context, I will have a look an that. Maybe a thread in the background could help here.

**Reply by Pocky0013** [reply]
@togrupe Regarding the range writing


It would be cool to be able to write to the table all 128 tracks as it is, but to be able to set a track range just for the reaper template.
In my way of working the 64 tracks of the avanti are in my config sheet, but often I only need to record 32 tracks live in my repaer

I don't know if I can explain well?

**Reply by Pocky0013** [reply]
the rest is wonderful

**Reply by togrupe** [reply]
The recording topic could be solved like that..

with that approach you can decide on a track bases what to record and what not..

<img width="451" alt="Bildschirmfoto 2023-04-28 um 10 09 28" src="https://user-images.githubusercontent.com/89751618/235092465-b768c10e-97f8-4550-ac87-74beca54c88b.png">

What do you think? And its easier to implement :)

**Reply by Pocky0013** [reply]
the idea is good. I wonder how to manage the routing in reaper if for example
![Capture](https://user-images.githubusercontent.com/111347509/235094387-0403e7e7-957d-4157-9821-692c09f6e53e.JPG)

I think that even if there are holes in the rec grid, it is better to keep the logical patch 1 to 1, 2 to 2,... 32 to 32

What do you think about it?

**Reply by togrupe** [reply]
Whenever possible I prefer 1:1 patches also for recording. I would get crazy about digital patching.

I derive the reaper patch from the Channel, technically possible would also be an extra mapping.

BTW I have also control about the "Record Arm" Button in Reaper. Based on that you can have it available in the Recording session, but "first" inactive, you can still decide to "arm" it.

Regarding your musical? Are you using  "Virtual Soundcheck" for Recording? or are you doing it manually via "Tie Lines"?

**Reply by Pocky0013** [reply]
I also prefer the 1:1 version
Then do we use the arm function or is the track simply not added? It's a personal question I think... I don't know what to think


I use the virtual soundchecks to do my sound checks without the artists. but I'm recording by patching the inputs directly into the i/o Card config

**Reply by togrupe** [reply]
We can add the "arm" feature at any time later. Should be also simple. The "Recording" = yes decides if the track is part of the session or not. This means by default an added Track is also "armed"

**Comment by togrupe** [comment]
Hey @Pocky0013 

here a stabilized version 

http://wp1054826.server-he.de/downloads/dlive-midi-tools/v2_2_0-alpha3-SNAPSHOT/dmt-v2_2_0-alpha3-SNAPSHOT-windows.zip

Changes:

- improved error handling in case of wrong ip format
- backgroundtask handling added. ("The process gets stuck" issue should be fixed)

**Comment by togrupe** [comment]
Couldn't resist continuing :D

here the first official release candidate v2.2.0-RC1 

http://wp1054826.server-he.de/downloads/dlive-midi-tools/v2_2_0-RC1/dmt-v2_2_0-RC1-windows.zip

Changes:

- Safe, Default and Director Button added to save/restore GUI setting
<img width="684" alt="Bildschirmfoto 2023-04-28 um 01 48 50" src="https://user-images.githubusercontent.com/89751618/235013835-2906394f-0411-475a-9d5c-7b4a0f509919.png">

**Reply by togrupe** [reply]
One more update...:)

here the second official release candidate v2.2.0-RC2

http://wp1054826.server-he.de/downloads/dlive-midi-tools/v2_2_0-RC2/dmt-v2_2_0-RC2-windows.zip

Changes:

- version for config.json introduced
- midi port is now also persisted

**Reply by Pocky0013** [reply]
MANNNN !  you are amazing.
next week, I will have to create 60 scenes on my console from an excel file with microphone opening and closing for a musical. Your software will save me so many mistakes and make my life easier.

**Reply by togrupe** [reply]
Thanks for YOUR Testing Support. So cool. I have so many more ideas. Did you have the chance to see  the Wiki page? 
https://github.com/togrupe/dlive-midi-tools/wiki

**Reply by Pocky0013** [reply]
I just went to see. Your ideas are super cool. I think the challenge is going to be to keep it simple and effective. But it's true that silly things like HPFs or DCAs take a bit of time to do on the console.
![Capture](https://user-images.githubusercontent.com/111347509/235096313-a7f82a75-ab54-4583-84f5-c8224e458667.JPG)
With formulas I filled in the given sheet. I think it's still not too complicated to send to customers as it is. To integrate more information, I think I should have a client version and a technician version.

**Reply by togrupe** [reply]
By the way.. you can extend the spreadsheet with extra information columns as you want e.g person, mic, stand, cable, etc...the only thing you shouldn't do is changing the main head column titles (Channel, Name, Color, Mute), because they are used to read the table.

**Reply by togrupe** [reply]
> I think the challenge is going to be to keep it simple and effective

I totally agree, I also want it as simple as possible to use. It should not replace Director as tool. DCA Assignment would be great, the logic is already implemented,  but unfortunately "NRPN is currently not supported from the midi library"

**Comment by togrupe** [comment]
Hey @Pocky0013 

the next evolution is available: v2_2_0-RC3

http://wp1054826.server-he.de/downloads/dlive-midi-tools/v2_2_0-RC3/dmt-v2_2_0-RC3-windows.zip

Please make sure to use the latest available spreadsheet (with version 4)

New features:
- recording flag introduced
- record arm flag introduced
- template spreadsheet updated (v4)

**Reply by togrupe** [reply]
Arm Feature
![grafik](https://user-images.githubusercontent.com/89751618/235166110-ff920ab8-340a-4d6f-8de0-bcac1f1e678a.png)

How the patching(1:1) looks like... 
![grafik](https://user-images.githubusercontent.com/89751618/235166441-b7d221cb-e91e-4896-9bb4-0f102481e602.png)

**Comment by togrupe** [comment]
@Pocky0013 

Just finished my dlive tests.

The new version v2.2.0 is published, Please feel free to download it. 

Thank you so much for you support.

Have a great weekend.

Tobi

**Reply by Pocky0013** [reply]
VERY NICE ! 
it works perfectly for me too!

**Reply by Pocky0013** [reply]
I just noticed that the software now takes longer to open than in the first versions. Nothing serious, but it's surprising. Have you noticed this at home too?

**Reply by togrupe** [reply]
> I just noticed that the software now takes longer to open than in the first versions. Nothing serious, but it's surprising. Have you noticed this at home too?

I didn´t notice it yet. 

Is it only the starting process till the window is open? or the whole writing process. I added a bit more logging at the beginning, but it should not be noticeable.

How many seconds? is it significantly higher?

**Reply by Pocky0013** [reply]
on my computer at work, I don't have this problem. So it shouldn't come from the software.

**Comment by togrupe** [comment]
@Pocky0013 

Hey, how was your musical? did every go well?

Are you ready for the next level? ;)

I have implemented new Features:

- Gain Support
- DCA Support
- Fader Level Support
- HPF On Support (only dLive)
- HPF Value Support (only dLive)

<img width="1162" alt="Bildschirmfoto 2023-05-07 um 18 01 11" src="https://user-images.githubusercontent.com/89751618/236688812-cb6c3f78-0248-40f8-8c8b-726f1412e324.png">

<img width="767" alt="Bildschirmfoto 2023-05-07 um 18 01 50" src="https://user-images.githubusercontent.com/89751618/236688823-25d7caa5-b6e1-4a68-8cab-91e2688ee308.png">


I have a new alpha version (v2.3.0-alpha4) available. Are you interested in testing it?

Tobi

**Reply by Pocky0013** [reply]
Hello,
I was able to program the mixer via excel for all 60 scenes. Fast, efficient and I'm sure I didn't make any "copy" errors. Now the shows will be from mid-May to the end of November. This musical comedy will take me the year ;-)

So cool! With pleasure that I test the new version on the Avantis and that I give you a feedback.

You were able to add the sending of DCAs !?!? this is really cool. But we still don't have the possibility at the firmware level to modify the name and the color, is that right? Either way, I'm looking forward to trying it out.

**Reply by togrupe** [reply]
@Pocky0013 

Here we go...

https://liveworks-vt.de/downloads/dlive-midi-tools/v2_3_0-alpha4/dmt-v2_3_0-alpha4-windows.zip

Looking forward for your feedback.

**Reply by togrupe** [reply]
>  But we still don't have the possibility at the firmware level to modify the name and the color, is that right?

Are you talking about the "DCA Names & Colors" ? if so, no that's unfortunately not possible.

**Reply by Pocky0013** [reply]
Yesterday was a busy day, I didn't have time to do the tests. I watch weekend when I have the console with me again

**Comment by togrupe** [comment]
@Pocky0013 

just added also the "Mute Group" Support. But it seems this feature is unfortunately not working for Avantis, at least it didn't work for me with "Avantis Director"

<img width="409" alt="Bildschirmfoto 2023-05-07 um 21 42 57" src="https://user-images.githubusercontent.com/89751618/236699365-f6e963ac-7e9b-4340-97c2-2c6572293e57.png">

But here the latest version (alpha5)
https://liveworks-vt.de/downloads/dlive-midi-tools/v2_3_0-alpha5/dmt-v2_3_0-alpha5-windows.zip

**Comment by Pocky0013** [comment]
in the meantime, what do you think of classifying the checkboxes so that it is more clear and intuitive? (Into the software)
For example : 
![Capture](https://user-images.githubusercontent.com/111347509/237016245-a782dd3b-5c8c-4941-8ddc-d25a71330d83.JPG)

**Reply by togrupe** [reply]
@Pocky0013 
Was a bit tricky...

but we have a new baby :)

https://liveworks-vt.de/downloads/dlive-midi-tools/v2_3_0-alpha7/dmt-v2_3_0-alpha7-windows.zip

<img width="902" alt="Bildschirmfoto 2023-05-10 um 04 29 11" src="https://github.com/togrupe/dlive-midi-tools/assets/89751618/125bb876-b66f-4ca5-91d2-191563770189">

What do you think?

**Reply by Pocky0013** [reply]
MANNN it's a masterpiece ! 
it is very clear and easy to understand

**Reply by togrupe** [reply]
Your idea was great...

btw.. we can organize the groups as we want it.. 

Did you have the chance to test it with the console?

I did my tests with both directors. 

If you want, you can also find me on facebook or instagram.

**Reply by Pocky0013** [reply]
this time there is no good news. What doesn't work on Avantis Director doesn't actually work on the console. I actually can't have the HPF, HPF Value and mute groups. On the other hand, the fader level and the DCAs work well. 
But it's consistent with what you saw


there is just a problem with the construction in the excel file. In the fader level column if I change to another value and put -inf back, it create a formula (=-inf) and the data validation doesn't work anymore. no problem for me, because I create my own excel file, but the file you provided has this problem.

**Reply by togrupe** [reply]
I think your observation matches to my Observations (with Avantis Director), „HPF On, HPF Value and the Mute Groups aren‘t Working, because the api doesnt support it.

fader, dcas worked for me as well.

Did you See the Message Box when you select „Avantis“ console, there should be a popup Message Like, „the following Features are Not Supportet…“

It could be, because of the settings persistence (config.json file) 
 you did not See the Message. 

Anyway, I will try to „Blend out“ the not Supported Features for Avantis and also Show the hint inmediately when the application starts.


Do you use Microsoft excel or libreoffice?
And which Version?

I will Check This issue, was it on a specific Channel? Or on all Channels?

**Reply by Pocky0013** [reply]
yes I see the message and it comes every time I change between Dlive or Avantis.
I use Microsoft excel, the last version coming with office 365. 

It was on all channels

**Reply by togrupe** [reply]
@Pocky0013 

Thanks for sharing your observations and the feedback.

Good that you test with Excel, I use Libreoffice on my Mac , then we have a good test coverage.

I fixed the spreadsheet issue, instead of "-inf" we now use "-99" that's better to handle. I found a way to disable the Avantis Checkboxes. This should make it even more clear that this is currently not possible.

Here you can find the first Release Candidate v2.3.0-RC1

https://liveworks-vt.de/downloads/dlive-midi-tools/v2_3_0-RC1/dmt-v2_3_0-RC1-windows.zip

https://liveworks-vt.de/downloads/dlive-midi-tools/v2_3_0-RC1/dmt-v2_3_0-RC1-macos.zip

**Comment by Pocky0013** [comment]
it's very clean to disable the checkboxes like you did !
Your copyright information is missing ;-) After that you can update the homepage of the project and we can do some facebook post to promote your work! Because I think it can be useful to many technicians.

**Reply by togrupe** [reply]
The page is already updated and prepared.
when we merge to Main, it becomes active. 

The 2.3.0 Download link is obviously not yet working.


https://github.com/togrupe/dlive-midi-tools/tree/release/2.3.0

**Reply by togrupe** [reply]
@Pocky0013 

here the "speed waggon" RC10 :)

https://liveworks-vt.de/downloads/dlive-midi-tools/v2_3_0-RC10/dmt-v2_3_0-RC10-windows.zip

Tested as much as possible with dlive.

**Reply by Pocky0013** [reply]
I did the same with the avantis. Before we were at 58 seconds, now 13 seconds! once again it's a master kill

**Reply by togrupe** [reply]
Could you please Review the Updated and prepared README.

https://github.com/togrupe/dlive-midi-tools/tree/release/2.3.0

if you give Green Light as well we can Release it.

**Reply by Pocky0013** [reply]
everything seems ok to me.
This morning I sent a show and I got an error. I had put a "é" in a name column. It's completely normal that it doesn't work. But should we put a note about it in the README?

**Reply by togrupe** [reply]
Very Good idea about the hint
